src/js/services/image.service.js

```diff
@@ -23,21 +23,20 @@
             }
 
             function createImagesArray(data){
-                var farm, server, id, title, secret, picUrl, currentPhoto;
+                var farm, server, id, title, secret, picUrl;
                 var images = [];
 
-                for(var i = 0; i < data.photos.photo.length; i++) {
-                    currentPhoto = data.photos.photo[i];
+                angular.forEach(data.photos.photo, function (value) {

```

Comment:
Again, name the "item" variable to something representing the data isn't referencing.